### PR TITLE
*: fix the error message it is an int type

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
In JDBC test, the error message for int type should use word 'integer' instead of 'int'. This pr fix it.

### What is changed and how it works?
The error message will look like "Incorrect integer value: 'a' for column 'i' at row 1".

### Check List 

Tests 

 - Unit test

Side effects

 - Increased code complexity

PTAL @coocood @XuHuaiyu

### CODE REVIEW

File: types/etc.go [hunk 1]
```diff
@@ -189,6 +189,15 @@ func TypeToStr(tp byte, cs string) (r string) {
 	return ts
 }
 
+// TypeToStrForErr converts a field to a string for the error messages.
+func TypeToStrForErr(tp byte) string {
+	ts := type2Str[tp]
+	if tp == mysql.TypeLong {
```
Comment (XuHuaiyu) [inline types/etc.go]: what about mysql.TypeLonglong

Comment (jackysp) [reply types/etc.go]: Actually, I don't know the exact map from the type to its name in all MySQL error message. I think we can add more types when we meet the other compatibility issues in future.

Comment (zimulala) [reply types/etc.go]:
```
mysql> create table t (i bigint);
Query OK, 0 rows affected (0.02 sec)
mysql> insert into t values ('a');
ERROR 1366 (HY000): Incorrect integer value: 'a' for column 'i' at row 1
```

Comment (jackysp) [reply types/etc.go]: OK, I'll try all the types of MySQL.

**Comment by jackysp** [comment]
Due to MySQL does not have a consistent error message, this pr maybe nearly no help for the compatibility.

**Pull request closed without merging**